docs/building-blocks/extending-templates/objects-as-parameters.md

```diff
@@ -0,0 +1,319 @@
+                        "sourcePortRange": "[parameters('networkSecurityGroupsSettings').securityRules[copyIndex()].sourcePortRange]",
+                        "destinationPortRange": "[parameters('networkSecurityGroupsSettings').securityRules[copyIndex()].destinationPortRange]",
+                        "sourceAddressPrefix": "[parameters('networkSecurityGroupsSettings').securityRules[copyIndex()].sourceAddressPrefix]",
+                        "destinationAddressPrefix": "[parameters('networkSecurityGroupsSettings').securityRules[copyIndex()].destinationAddressPrefix]",
+                        "access":"[parameters('networkSecurityGroupsSettings').securityRules[copyIndex()].access]",
+                        "direction":"[parameters('networkSecurityGroupsSettings').securityRules[copyIndex()].direction]"
+                        }
+                  }
+            ],
+            "outputs": {}
+          }
+        }
+    }
+  ],          
+  "outputs": {}
+}
+
+```
+
+The corresponding parameter file looks like this:
+
+```json
+{
+    "$schema": "https://schema.management.azure.com/schemas/2015-01-01/deploymentParameters.json#",
+    "contentVersion": "1.0.0.0",
+    "parameters":{ 
+      "networkSecurityGroupsSettings": {
+      "value": {
+          "securityRules": [
+            {
+              "name": "RDPAllow",
+              "description": "allow RDP connections",
+              "direction": "Inbound",
+              "priority": 100,
+              "sourceAddressPrefix": "*",
+              "destinationAddressPrefix": "10.0.0.0/24",
+              "sourcePortRange": "*",
+              "destinationPortRange": "3389",
+              "access": "Allow",
+              "protocol": "Tcp"
+            },
+            {
+              "name": "HTTPAllow",
+              "description": "allow HTTP connections",
+              "direction": "Inbound",
+              "priority": 200,
+              "sourceAddressPrefix": "*",
+              "destinationAddressPrefix": "10.0.1.0/24",
+              "sourcePortRange": "*",
+              "destinationPortRange": "80",
+              "access": "Allow",
+              "protocol": "Tcp"
+            }
+          ]
+        }
+      }
+    }
+  }
+```
+
+## Try the template
+
+If you would like to experiment with these templates, follow these steps:
+
+1.	Go to the Azure portal, select the "+" icon, and search for the "template deployment" resource type. When you find it in the search results, select it.
+2.	When you get to the "template deployment" page, select the **create** button. This button opens the "custom deployment" blade.
+3.	Select the **edit template** button.
+4.	Delete the empty template in the right-hand pane.
+5.	Copy and paste the sample template into the right-hand pane.
+6.	Select the **save** button.
+7.	When you are returned to the "custom deployment" pane, select the **edit parameters** button.
```

Comment:
this sounds strange again, 'when you are returned'... how about 'When youa re back at'? or 'Once you are back at'?